subsys/bluetooth/Kconfig.scan

```diff
@@ -116,6 +116,73 @@ config BT_SCAN_MANUFACTURER_DATA_CNT
 
 endif
 
+config BT_SCAN_CONNECTION_ATTEMPTS_FILTER
+	bool "Connection attempts filter"
+	help
+	 Connection attempts filter. This filter limits the number of attempts
+	 to connect to the device in case of fail.
+
+if BT_SCAN_CONNECTION_ATTEMPTS_FILTER
+
+config BT_SCAN_CONNECTION_ATTEMPTS_DEVICE_CNT
+	int "Connection attempts filtered device count"
+	default 2
+	help
+	  The maximum number of the filtered devices by
+	  the connection attempts filter.
+
+config BT_SCAN_CONNECTION_ATTEMPTS_CNT
+	int "Connection attempts count"
+	default 2
+	help
+	 Connection attempts count. Defines how many times the device will
+	 try to reconnect if the previous attempt failed.
+endif
+
+if !BT_SCAN_CONNECTION_ATTEMPTS_FILTER
+
+config BT_SCAN_CONNECTION_ATTEMPTS_DEVICE_CNT
+	int
+	default 0
+	help
+	  The maximum number of the filtered devices by
+	  the connection attempts filter.
+
+config BT_SCAN_CONNECTION_ATTEMPTS_CNT
+	int
+	default 0
+	help
+	 Connection attempts count. Defines how many times the device will
+	 try to reconnect if the previous attempt failed.
+
+endif
+
+config BT_SCAN_BLACKLIST
+	bool "Scanning device blacklist"
+	help
+	  Scanning device blacklist. Scanning module ignores
+	  any devices which are on this list.
+
+if BT_SCAN_BLACKLIST
+
+config BT_SCAN_BLACKLIST_DEVICE_CNT
+	int "Blacklist maximum device count"
+	default 2
```

Comment:
The array size base on this config. It cannot be simplify in this way

Reply:
The size of the array would be set to 0 by default value (in C code it will work in the same way as the current implementation).
The only risk is that user would change the value without enabling the functionality (that would lead to wasting RAM)